src/dune_rules/pkg_rules.ml

```diff
@@ -621,6 +623,109 @@ module Substitute = struct
 end
 
 module Run_with_path = struct
+  module Output : sig
+    type t
+
+    val io : t -> Process.Io.output Process.Io.t
+
+    val v
+      :  ?accepted_exit_codes:int Predicate.t
+      -> ?pkg_name:Dune_lang.Package_name.t
```

Comment:
we avoid optional arguments in new code - you can make `accepted_exit_codes` a labelled argument and `pkg_name` an option.